lib/events.js

```diff
@@ -84,8 +106,31 @@ EventEmitter.init = function() {
   }
 
   this._maxListeners = this._maxListeners || undefined;
+
+  if (opts && opts.autoCatch) {
+    this[kApply] = applyAndCatch;
+  }
 };
 
+function applyAndCatch(handler, that, args) {
+  const promise = Reflect.apply(handler, that, args);
+  if (promise && typeof promise.catch === 'function') {
```

Comment:
consider using thenable protocol here

```js
if (promise && typeof promise.then === 'function') {
  promise.then(undefined, (err) => {
    ...
  });
}
```

Reply:
Done! Can you please check the docs and see if I need to change some wording?